packages/cadl-ranch-specs/http/clients/authentication/main.cadl

```diff
@@ -0,0 +1,26 @@
+import "@cadl-lang/rest";
+import "@azure-tools/cadl-azure-core";
+
+using Cadl.Http;
+using Azure.Core.Foundations;
+
+@serviceTitle("ClientAuthentication")
+@serviceVersion("1.0.0")
+@server("http://localhost:3000", "TestServer endpoint")
+@doc("Illustrates clients generated with OAuth2 and ApiKey authentication.")
+
+@useAuth(OAuth2Auth<[OAuth2AuthFlow]> | ApiKeyAuth<ApiKeyLocation.header, "x-ms-api-key">)
+namespace Cadl.TestServer.ClientAuthentication;
+
+model OAuth2AuthFlow {
+    type: OAuth2FlowType.clientCredentials;
+    tokenUrl: "https://api.example.com/oauth2/token";
+    refreshUrl: "https://api.example.com/oauth2/refresh";
+    scopes: [
+        "https://api.example.com/.default"
+    ]
+}
+
+@route("/clients/authentication/get")
+@get
+op get(): OkResponse | ErrorResponse;
```

Comment:
Would need to add the `@scenario` decorator to mark the endpoint as a scenario.
I expect at least of have something like
- `authenticatedCall`: Expect the test to send a valid oauth2 token. 

and then not sure if we need that:
- `invalidAuthenticationCall` : Expect the test to send a invalid oauth2 token and fail the request and the client should handle fine.